sdk/src/main/java/co/omise/android/ui/AuthorizingPaymentActivity.kt

```diff
@@ -59,11 +60,17 @@ class AuthorizingPaymentActivity : AppCompatActivity() {
         supportActionBar?.title = uiCustomization.uiCustomization.toolbarCustomization?.headerText
             ?: getString(R.string.title_authorizing_payment)
 
-        if (verifier.verifyExternalURL(verifier.authorizedURL)) {
-            openDeepLink(verifier.authorizedURL)
+        val authUrl = verifier.authorizedURLString
+        // check for legacy payments that require web view
+        if (authUrl.endsWith("/pay")) {
+            setupWebView()
+        } else {
+            // Check if the URL needs to be opened externally
+            if (verifier.verifyExternalURL(verifier.authorizedURL)) {
+                openDeepLink(verifier.authorizedURL)
+            }
+            observeData()
```

Comment:
These lines are for authorization? How about moving these to a separate function like below. We might also combine `if..else { if }` as `if...else if`.

```
override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_authorizing_payment)

        supportActionBar?.title = uiCustomization.uiCustomization.toolbarCustomization?.headerText
            ?: getString(R.string.title_authorizing_payment)

        handlePaymentAuthorization()
}

private fun handlePaymentAuthorization() {
    val authUrl = verifier.authorizedURLString
 
    if (isLegacyPayment(authUrl)) {
        setupWebView()
    } else if (verifier.verifyExternalURL(verifier.authorizedURL)) {
         openDeepLink(verifier.authorizedURL)
    }

    observeData()
}
```